homeassistant/components/intent/timers.py

```diff
@@ -0,0 +1,772 @@
+"""Timer implementation for intents."""
+
+from __future__ import annotations
+
+import asyncio
+from dataclasses import dataclass
+from functools import cached_property
+import logging
+import time
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.const import ATTR_DEVICE_ID, ATTR_ID, ATTR_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import (
+    area_registry as ar,
+    config_validation as cv,
+    device_registry as dr,
+    intent,
+)
+from homeassistant.util import ulid
+
+from .const import (
+    ATTR_DATA,
+    ATTR_LANGUAGE,
+    ATTR_PAUSED,
+    ATTR_SECONDS_LEFT,
+    ATTR_START_HOURS,
+    ATTR_START_MINUTES,
+    ATTR_START_SECONDS,
+    EVENT_INTENT_TIMER_CANCELLED,
+    EVENT_INTENT_TIMER_FINISHED,
+    EVENT_INTENT_TIMER_STARTED,
+    EVENT_INTENT_TIMER_UPDATED,
+    TIMER_DATA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+TIMER_NOT_FOUND_RESPONSE = "timer_not_found"
+MULTIPLE_TIMERS_MATCHED_RESPONSE = "multiple_timers_matched"
+
+
+@dataclass
+class TimerInfo:
+    """Information for a single timer."""
+
+    id: str
+    """Unique id of the timer."""
+
+    name: str | None
+    """User-provided name for timer."""
+
+    seconds: int
+    """Total number of seconds the timer should run for."""
+
+    device_id: str | None
+    """Id of the device where the timer was set."""
+
+    task: asyncio.Task
+    """Background task sleeping until timer is finished."""
+
+    start_hours: int | None
+    """Number of hours the timer should run as given by the user."""
+
+    start_minutes: int | None
+    """Number of minutes the timer should run as given by the user."""
+
+    start_seconds: int | None
+    """Number of seconds the timer should run as given by the user."""
+
+    created_at: int
+    """Timestamp when timer was created (time.monotonic_ns)"""
```

Comment:
the downside of not using UTC seconds is that we cannot share this info with other consumers. Not sure if we want this? I guess we can add it later.